Library/Homebrew/cmd/list.rb

```diff
@@ -39,7 +39,7 @@ def list
       filtered_list
     elsif ARGV.named.empty?
       if ARGV.include? "--full-name"
-        full_names = Formula.installed.map(&:full_name).sort &tap_and_name_comparison
+        full_names = Formula.installed.map(&:full_name).sort(&tap_and_name_comparison)
```

Comment:
Please remove this change as it's otherwise unrelated to this PR, thanks!

Reply:
This change was actually prompted by a test failure I saw on my local:

      1) brew readall imports all Formulae for a given Tap
         Failure/Error:
           expect { brew "readall", "--aliases", "--syntax" }
             .to be_a_success
             .and not_to_output.to_stdout
             .and not_to_output.to_stderr
     
          expected #<Proc:0x00000104b03ee8@/usr/local/Homebrew/Library/Homebrew/test/cmd/readall_spec.rb:10> to be a success
     
       ...and:
     
          expected block to not output to stderr, but output "/usr/local/Homebrew/Library/Homebrew/cmd/list.rb:42: warning: `&' interpreted as argument prefix\n"
     # ./test/cmd/readall_spec.rb:10:in `block (2 levels) in <top (required)>'
     # ./test/support/helper/spec/shared_context/integration_test.rb:44:in `block (2 levels) in <top (required)>'
     # ./test/spec_helper.rb:96:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'

    Finished in 27.04 seconds (files took 1.98 seconds to load)
    198 examples, 1 failure

    Failed examples:

    rspec ./test/cmd/readall_spec.rb:2 # brew readall imports all Formulae for a given Tap

I also saw failures in the Travis CI build when I submitted an earlier PR without this change, but they went away when I re-submitted with this addition.  I'm not sure what it is about my change in `readall` that triggers a warning from `list.rb`.  I can investigate if you'd like.